mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -103,18 +103,27 @@ pub const WATCHER_MESSAGE_SENT_LOG: &str = "Watcher message sent...";
 pub const MAKER_PAYMENT_SPENT_BY_WATCHER_LOG: &str = "Maker payment is spent by the watcher...";
 
 #[cfg(not(target_arch = "wasm32"))]
-pub fn stats_taker_swap_dir(ctx: &MmArc) -> PathBuf { ctx.dbdir().join("SWAPS").join("STATS").join("TAKER") }
+pub fn stats_taker_swap_dir(ctx: &MmArc) -> PathBuf {
+    #[cfg(not(feature = "new-db-arch"))]
+    return ctx.dbdir().join("SWAPS").join("STATS").join("TAKER");
+    #[cfg(feature = "new-db-arch")]
+    return ctx.global_dir().join("SWAPS").join("STATS").join("TAKER");
```

Comment:
Personally, I would use `if cfg!(feature = "..")` to avoid having to disable/enable the `new-db-arch` feature just to get LSP working while working on these parts.

Reply:
> If I was working on this task I wouldn't like to mess with features all the time while working on it

me too i think.
i was trying this new method since we can use it to incrementally roll the new arch into the codebase rather than having a very big PR. i think i like it though :D

Reply:
[7bcd119](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2398/commits/7bcd1192cffac4ccbc188a1dc447b89b0bbd0d02)